tests/ui/consts/refs-to-cell-in-final.rs

```diff
@@ -28,10 +28,8 @@ impl Drop for JsValue {
     fn drop(&mut self) {}
 }
 const UNDEFINED: &JsValue = &JsValue::Undefined;
-//~^ WARNING: mutable pointer in final value of constant
-//~| WARNING: this was previously accepted by the compiler but is being phased out
 
-// In contrast, this one works since it is being promoted.
+// Here's a variant of the above that uses promotion instead of the "outer scope rule" rule.
```

Comment:
needs rebase on the typo changes done at the PR this is based on, but can be done after we get a decision from the team about this